apps/s_client.c

```diff
@@ -2408,11 +2408,16 @@ int s_client_main(int argc, char **argv)
         break;
     case PROTO_IRC:
         {
-            int numeric;
+            int numeric, pos, foundit = 0;
+            char *buf;
+            char command[11] = { 0 };
             BIO *fbio = BIO_new(BIO_f_buffer());
 
+            BIO_printf(bio_c_out, "[sending plaintext] CAP LS 302\\r\\n"
+                       "STARTTLS\\r\\n\n");
             BIO_push(fbio, sbio);
-            BIO_printf(fbio, "STARTTLS\r\n");
+            BIO_printf(fbio, "CAP LS 302\r\n"
+                       "STARTTLS\r\n");
```

Comment:
Can you explain why this is needed? There should be no need to list the IRC server capabilities first, it should always reply to STARTTLS if if supports it, see: https://ircv3.net/specs/extensions/tls-3.1.html

Reply:
I removed it since it does not seem to be strictly required